ratatui/benches/main/constraints.rs

```diff
@@ -0,0 +1,59 @@
+use std::hint::black_box;
+use std::rc::Rc;
+
+use criterion::{Criterion, criterion_group};
+use ratatui::layout::Constraint::{Fill, Length, Max, Min, Percentage, Ratio};
+use ratatui::layout::{Direction, Layout, Rect};
+
+const SPLIT_BY: u16 = 10;
+
+fn generate_layout(constraint: &str, area: Rect) -> Rc<[Rect]> {
+    Layout::default()
+        .direction(Direction::Vertical)
+        .constraints({
+            let mut cs = Vec::new();
+            for _ in 0..SPLIT_BY {
+                match constraint {
+                    "Fill" => {
+                        cs.push(Fill(1));
+                    }
+                    "Length" => {
+                        cs.push(Length(area.width / SPLIT_BY));
+                    }
+                    "Max" => {
+                        cs.push(Max(area.width / SPLIT_BY));
+                    }
+                    "Min" => {
+                        cs.push(Min(area.width / SPLIT_BY));
+                    }
+                    "Percentage" => {
+                        cs.push(Percentage(100 / SPLIT_BY));
+                    }
+                    "Ratio" => {
+                        cs.push(Ratio(1, SPLIT_BY.into()));
+                    }
+                    _ => todo!(),
+                }
+            }
+            cs
+        })
+        .split(area)
+}
+
+fn constraints_render(criterion: &mut Criterion) {
+    for size in [16, 64, 255] {
+        let mut group = criterion.benchmark_group(format!("constraints {size}x{size}"));
+        group.sample_size(1000);
```

Comment:
What's the purpose of the custom sample size. If the default isn't good, then it's worth adding a comment about why.

Reply:
I don't know why that has the sample size set either (. There's got to be some process to working out the correct sample size, but when that happens it's reasonable to document things so that the findings can be repeated later. I'm fine with leaving it out for now too.